source/win32_tinyengine.c

```diff
@@ -163,6 +391,819 @@ Win32MainWindowCallback(HWND Window, UINT Message, WPARAM WParam, LPARAM LParam)
+                        } break;
+
+                        ///
+
+                        case VK_KANA:
+                        {
+                            // NOTE(hayden): Shared VKCode
+                            Win32Keyboard[KEY_MODE_KANA].WMDown = IsDown;
+                            Win32Keyboard[KEY_MODE_HANGEUL].WMDown = IsDown;
+                            Win32Keyboard[KEY_MODE_JUNJA].WMDown = IsDown;
+                        } break;
+
+                        case VK_JUNJA:
+                        {
+                            Win32Keyboard[KEY_MODE_JUNJA].WMDown = IsDown;
+                        } break;
+
+                        case VK_FINAL:
+                        {
+                            Win32Keyboard[KEY_MODE_FINAL].WMDown = IsDown;
+                        } break;
+
+                        case VK_KANJI:
+                        {
+                            // NOTE(hayden): Shared VKCode
+                            Win32Keyboard[KEY_MODE_HANJA].WMDown = IsDown;
+                            Win32Keyboard[KEY_MODE_KANJI].WMDown = IsDown;
+                        } break;
+
+                        ///
+
+                        case VK_ATTN:
+                        {
+                            Win32Keyboard[KEY_ATTN].WMDown = IsDown;
+                        } break;
+
+                        case VK_CRSEL:
+                        {
+                            Win32Keyboard[KEY_CRSEL].WMDown = IsDown;
+                        } break;
+
+                        case VK_EXSEL:
+                        {
+                            Win32Keyboard[KEY_EXSEL].WMDown = IsDown;
+                        } break;
+
+                        case VK_EREOF:
+                        {
+                            Win32Keyboard[KEY_EREOF].WMDown = IsDown;
+                        } break;
+
+                        case VK_PLAY:
+                        {
+                            Win32Keyboard[KEY_PLAY].WMDown = IsDown;
+                        } break;
+
+                        case VK_ZOOM:
+                        {
+                            Win32Keyboard[KEY_ZOOM].WMDown = IsDown;
+                        } break;
+
+                        case VK_NONAME:
+                        {
+                            Win32Keyboard[KEY_NONAME].WMDown = IsDown;
+                        } break;
+
+                        case VK_PA1:
+                        {
+                            Win32Keyboard[KEY_PA1].WMDown = IsDown;
+                        } break;
+                    }
+
+                    // getting a human-readable string
+                    //UINT Key = (ScanCode << 16) | (IsE0 << 24);
+                    //char Buffer[512] = {0};
+                    //GetKeyNameTextW((LONG)Key, Buffer, 512);
+                }
+            }
+        } break;
+
+        case WM_MOUSEMOVE:
+        {
+            // Cursor
+            Win32Mouse.X = LParam & 0xFFFF;
```

Comment:
Maybe just leave a comment saying where these are from (if i remember correctly this is just a define)

Reply:
It was a macro defined in `<windowsx.h>`. Since this was the only part of the code that would have used that header, I just expanded the macro out here -- should I include the header after all, or maybe just pull out the define and put it in `win32_platform.h`? In any case, I've put a note there for now